src/QsCompiler/Core/Dependencies.fs

```diff
@@ -29,191 +31,168 @@ type BuiltIn = {
+    /// Returns true if the given attribute marks the corresponding declaration as unit test.
+    static member MarksTestOperation (att : QsDeclarationAttribute) = att.TypeId |> function
+        | Value tId -> tId.Namespace.Value = BuiltIn.Test.FullName.Namespace.Value && tId.Name.Value = BuiltIn.Test.FullName.Name.Value
         | Null -> false
 
 
     // hard dependencies in Microsoft.Quantum.Core
 
     static member Length = {
-        Name = "Length" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New)
+        FullName = {Name = "Length" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Function (TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New))
     }
 
     static member RangeReverse = {
-        Name = "RangeReverse" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty    
+        FullName = {Name = "RangeReverse" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Function (TypeParameters = ImmutableArray.Empty)
     }
 
     static member Attribute = {
-        Name = "Attribute" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+        FullName = {Name = "Attribute" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
     static member EntryPoint = {
-        Name = "EntryPoint" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+        FullName = {Name = "EntryPoint" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
     static member Deprecated = {
-        Name = "Deprecated" |> NonNullable<string>.New
-        Namespace = BuiltIn.CoreNamespace
-        TypeParameters = ImmutableArray.Empty
+        FullName = {Name = "Deprecated" |> NonNullable<string>.New; Namespace = BuiltIn.CoreNamespace}
+        Kind = Attribute
     }
 
     static member Test = {
-        Name = "Test" |> NonNullable<string>.New
-        Namespace = BuiltIn.DiagnosticsNamespace
-        TypeParameters = ImmutableArray.Empty
+        FullName = {Name = "Test" |> NonNullable<string>.New; Namespace = BuiltIn.DiagnosticsNamespace}
+        Kind = Attribute
     }
 
     // hard dependencies in Microsoft.Quantum.Canon
 
     static member NoOp = {
-        Name = "NoOp" |> NonNullable<string>.New
-        Namespace = BuiltIn.CanonNamespace
-        TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New)
+        FullName = {Name = "NoOp" |> NonNullable<string>.New; Namespace = BuiltIn.CanonNamespace}
+        Kind = Operation (TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New), IsSelfAdjoint = false)
```

Comment:
```suggestion
        Kind = Operation (TypeParameters = ImmutableArray.Create("T" |> NonNullable<string>.New), IsSelfAdjoint = true)
```
It really is self-adjoint, and this information is used only within the compiler, so nothing will break if there is a discrepancy between the library and here.